### ISSUE TITLE
Linux PS3NetSrv

### ISSUE THREAD

**Issue by wes1993** [aldostools/webMAN-MOD#434]
Hello, 
when i run the PS3NetSRV this is the log:
```
ps3netsrv build 20200529 (mod by aldostools)
Path: /samba/wes93/PS3

Host IP #1: 192.168.0.15:38008
Host IP #2: 172.30.32.1:38008
Host IP #3: 172.17.0.1:38008
Waiting for client...
[0] Connection from 192.168.0.102
open dir /GAMES
open dir /PS3ISO
free(): invalid pointer
Annullato
```

The PS3 can see the games but the server crash so won't work.. :-(

with this version is ok:

```
ps3netsrv build 20170310 (mod by aldostools)
Path: /samba/wes93/PS3

Waiting for client...
Connection from 192.168.0.102
```
Some suggestion on how to fix this?
Thanks

# Update:
I have compiled this version but now i have this error:

```
root@server:/ScriptStefano# ./ps3netsrv_linuxX86_64 /samba/wes93/PS3 38008 192.168.0.*
ps3netsrv build 20200708 (mod by aldostools)
Path: /samba/wes93/PS3

Host IP #1: 192.168.0.15:38008
Host IP #2: 172.30.32.1:38008
Host IP #3: 172.17.0.1:38008
Waiting for client...
[0] Connection from 192.168.0.102
open dir /GAMES
open dir /PS3ISO
open dir /PSXGAMES
open dir /GAMES
open dir /PS3ISO
open dir /PSXGAMES
building virtual iso...
viso error: failed to create virtual iso
open error on "/GAMES/BCES01175-[Uncharted 3 - L'Inganno Di Drake]" (viso=1)
```

I have also this errror if i reboot my server:
```
Oct 30 11:56:54 server kernel: [ 1674.405691] ps3netsrv_linux[11935]: segfault at 10 ip 0000000000465bb1 sp 00007fcd69d1f448 error 4 in ps3netsrv_linuxX86_64[400000+10b000]
```
With old version: 
```
ps3netsrv build 20170310 (mod by aldostools)
Path: /samba/wes93/PS3

Waiting for client...
Connection from 192.168.0.102
building virtual iso...
open /GAMES/BCES01175-[Uncharted 3 - L'Inganno Di Drake]
Sorry no support for 666 files. I have the feeling that your game is about to crash ^_^
```

What mean 666 files?

**Comment by aldostools** [comment]
The error about 666 files occurs when your game contains split files .66600, .66601, .etc.
You need to merge these files into a single file. These files are created when large files (>4GB) are copied to USB FAT32.

You can try compiling this version (it's the latest code except for VisoFile module that is the old code)
[ps3netsrv_20201030.zip](https://github.com/aldostools/webMAN-MOD/files/5465199/ps3netsrv_20201030.zip)

**Comment by wes1993** [comment]
Thanks a lot for your reply:)
I will try if you want I can send the compiled version here.
One question how can I merge this files?

**Comment by aldostools** [comment]
On Windows you can use a CMD command line like: 

`copy /b file.66600+file.66601+file.66602 file.psarc`

On linux you use something like:
`cat file.66600 file.66601 file.66602 > file.psarc`

**Comment by wes1993** [comment]
Really Thanks a lot Again
Stefano

**Comment by wes1993** [comment]
I have compiled the server and i have this log during compiling:
![image](https://user-images.githubusercontent.com/52070589/97709835-3f693c80-1abb-11eb-827f-4205d8eaab3e.png)

is normal?
Stefano

**Comment by aldostools** [comment]
These are compilation warnings, not critical errors.
We are mixing modules from 2 versions, so it's "normal". Although they should be fixed later if the binary works fine.

**Comment by wes1993** [comment]
ok i will update you :-D

**Comment by wes1993** [comment]
Perfect Now Seems OK:
![image](https://user-images.githubusercontent.com/52070589/97712618-f915dc80-1abe-11eb-8414-d1c12bd6af3a.png)

But my PS3 Stuck so if is possible a good thing is to don't mount the game if 666 file stay in the game

**Comment by aldostools** [comment]
If you already merged the files, you must delete them (or move them outside the game if you want to keep the files),

**Comment by wes1993** [comment]
Yes but if someone download the game without seen that sare splitted i this the best thing is to not mout the game.
Just an error in the server log and if possibile in the notification of PS3 so we know the problem and merge the file
 Thanks again
Stefano ;-D

**Comment by Joonie86** [comment]
No further feedback
